dbt_sugar/core/task/base.py

```diff
@@ -1,10 +1,162 @@
 """API definition for Task-like objects."""
 import abc
+import os
+from pathlib import Path
+from typing import Any, Dict, List
+
+from dbt_sugar.core.clients.yaml_helpers import open_yaml, save_yaml
+from dbt_sugar.core.logger import GLOBAL_LOGGER as logger
+
+COLUMN_NOT_DOCUMENTED = "No description for this column."
+MODEL_NOT_DOCUMENTED = "No description for this model."
 
 
 class BaseTask(abc.ABC):
     """Sets up basic API for task-like classes."""
 
+    def __init__(self) -> None:
+        self.repository_path = Path().absolute()
+        self.dbt_definitions: Dict[str, str] = {}
+        self.save_all_descriptions()
+
+    def get_column_description_from_dbt_definitions(self, column_name: str) -> str:
```

Comment:
Do you think it makes sense to have all of these methods in the base task? The `BaseTask` is supposed to be an abstract class so to me the concrete implementations should live in `DocumentTask` no?

It is my understanding that you put abstractmethods in an abstract class to define the minimum viable methods (generally the ones expected by the task factory or the rest of your system) in there as non implemented ones and then you flesh out the concrete classes with whatever you need to have in there.

Have I been thinking about this all wrong?

Reply:
these methods are going to be used all the time in different tasks because they are the basics commands, that is the reason that was in a separate script if we add them into DocumentTask is going to be a lot of replication.

Reply:
Yeah I get that. If we are for sure going to need them in **all** of the tasks then I guess it can stay there. We might want to break down the base classes more into progressively more focused stuff when we come up with more tasks down the line.